scripts/patch-spl-crates-for-anchor.sh

```diff
@@ -0,0 +1,59 @@
+spl_assocated_token_account_version=
+spl_memo_version=
+spl_token_version=
+spl_token_2022_version=
+spl_tlv_account_resolution_verison=
+spl_transfer_hook_interface_version=
+
+get_spl_versions() {
+    declare spl_dir="$1"
+    spl_assocated_token_account_version=$(readCargoVariable version "$spl_dir/associated-token-account/program/Cargo.toml")
+    spl_memo_version=$(readCargoVariable version "$spl_dir/memo/program/Cargo.toml")
+    spl_token_version=$(readCargoVariable version "$spl_dir/token/program/Cargo.toml")
+    spl_token_2022_version=$(readCargoVariable version "$spl_dir/token/program-2022/Cargo.toml"| head -c1) # only use the major version for convenience
```

Comment:
Sorry, just for my own knowledge, what's the extra convenience exactly?

Reply:
Haha, I knew I should have gone back to grab the error to include on this PR. The convenience is that it actually works. Otherwise, it triggers this error:
```
error: failed to select a version for `spl-token-2022`.
    ... required by package `solana-account-decoder v2.0.0 (/Users/tyeraeulberg/Documents/Solana/solana/account-decoder)`
    ... which satisfies dependency `solana-account-decoder = "=2.0.0"` of package `anchor-client v0.29.0 (/Users/tyeraeulberg/Documents/Solana/solana/target/downstream-projects-anchor/anchor/client)`
    ... which satisfies path dependency `anchor-client` (locked to 0.29.0) of package `anchor-cli v0.29.0 (/Users/tyeraeulberg/Documents/Solana/solana/target/downstream-projects-anchor/anchor/cli)`
versions that meet the requirements `=2.0.1` are: 2.0.1

all possible versions conflict with previously selected packages.

  previously selected package `spl-token-2022 v2.0.2 (/Users/tyeraeulberg/Documents/Solana/solana/target/downstream-projects-anchor/spl/token/program-2022)`
    ... which satisfies dependency `spl-token-2022 = "^2.0.2"` of package `anchor-spl v0.29.0 (/Users/tyeraeulberg/Documents/Solana/solana/target/downstream-projects-anchor/anchor/spl)`

failed to select a version for `spl-token-2022` which could resolve this conflict
```

This is because the agave repo is on token-2022 v2.0.1, but the SPL repo has already bumped to v2.0.2.
It's actually an argument for using just major versions for all the SPL crates being patched (which I did have originally), as that will allow more flexibility in SPL before this job breaks again.

Reply:
Ah gotcha, thanks for the explanation! "things working" is the ultimate convenience :grinning:

Reply:
Or it's an argument for doing the double patching that you mentioned on discord, but I really didn't wanna...

Reply:
haha *same*